### ISSUE TITLE
Allow regex in `vue/no-restricted-class`

### ISSUE THREAD

**Issue by jd-solanki** [vuejs/eslint-plugin-vue#1877]
**What rule do you want to change?**
[vue/no-restricted-class](https://eslint.vuejs.org/rules/no-restricted-class.html)

**Does this change cause the rule to produce more or fewer warnings?**
No

**How will the change be implemented? (New option, new default behavior, etc.)?**
Allow a regex pattern just like we did [here](https://github.com/vuejs/eslint-plugin-vue/blob/master/lib/rules/no-restricted-custom-event.js#L30)

**Please provide some example code that this change will affect:**

```vue
<div class="ml-2" />
```

**What does the rule currently do for this code?**
Right now to restrict the classes which are not RTL friendly I have to write long list of them:
```js
'vue/no-restricted-class': ['error', 'ml-1', 'ml-2', /* And more... */],
```

**What will the rule do after it's changed?**
Once this rule accept regex pattern we don't have to write long list, just below is enough:
```js
'vue/no-restricted-class': ['error', '/^(m|p)(l|r)-/'],
```

**Additional context**
None

**Comment by FloEdelmann** [comment]
PRs welcome!
